controllers/servicebinding_controller.go

```diff
@@ -145,14 +145,26 @@ func (r *ServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			}
 
 			if binding.LastOperation.Type != smTypes.CREATE || binding.LastOperation.State == smTypes.SUCCEEDED {
-				// store secret unless binding is still being created or failed during creation
-				if err := r.storeBindingSecret(ctx, serviceBinding, binding, log); err != nil {
+				secretExist, err := r.isSecretAlreadyExist(ctx, serviceBinding)
+				if err != nil {
```

Comment:
maybe validate the error 
if apierrors.IsAlreadyExists(err) {
  log.Info(fmt.Sprintf("secret with name %s already exist)
} 
less code in reconciler and no need for isSecretAlreadyExists

Reply:
you can't ignore in recovery it still can be someone else secret